File: 485. Max Consecutive Ones.cpp
class Solution {
public:
     // count:O(N)time, O(1)space, 61.25%, ok
    int findMaxConsecutiveOnes(vector<int>& nums) {
        if(nums.empty()) return 0;  // bug: corner case
        int cnt = (nums[0] == 1);
        int ret = cnt;
        for(int i = 1; i < nums.size(); i++){
            if(nums[i]) cnt++;
            else {
                ret = max(ret, cnt);
                cnt = 0;
            }
        }
        return max(ret, cnt);
    }
};

File: 485. Max Consecutive Ones.py
class Solution(object):
    
    # Brute force: O(N) time, O(1) space, 61.01%
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        count, maxCnt = int(nums[0] == 1), 0
        for i in xrange(1, len(nums)):
            if nums[i] == 1:
                count += 1
            else:
                count, maxCnt = 0, max(count, maxCnt)
        return max(maxCnt, count)
                